core/src/main/scala/org/apache/gearpump/cluster/AppDescription.scala

```diff
@@ -142,4 +138,25 @@ case class ExecutorContext(
 case class ExecutorJVMConfig(
     classPath: Array[String], jvmArguments: Array[String], mainClass: String,
     arguments: Array[String], jar: Option[AppJar], username: String,
-    executorAkkaConfig: Config = ConfigFactory.empty())
\ No newline at end of file
+    executorAkkaConfig: Config = ConfigFactory.empty())
+
+sealed abstract class ApplicationStatus(val status: String, val isTerminal: Boolean)
```

Comment:
We can have a `TerminalStatus` extending `ApplicationStatus` for pattern matching rather than checking the flag every time.